backend/tet/events/services.py

```diff
@@ -57,29 +58,30 @@ def _get_event_and_raise_for_unauthorized(self, user, event_id):
         See `test_services.py` to understand the access logic better.
         """
         event = self.client.get_event(event_id)
-        LOGGER.warning(event)
+
+        if settings.NEXT_PUBLIC_MOCK_FLAG:
+            return event
+
         if self._is_city_employee(user):
             if not _user_matches(event, user):
+                # TODO audit log?
                 LOGGER.warning(
-                    f"User {user.email} was denied unauthorized access to event {event['id']}"
+                    f"User {user.pk} was denied unauthorized access to event {event['id']}"
                 )
-                _raise_unless_mocked()
+                _raise_permission_denied()
         else:
             LOGGER.warning(
                 "Authorization not implemented for company users (suomi.fi login)"
             )
-            _raise_unless_mocked()
+            _raise_permission_denied()
 
         return event
 
     def _filter_events_for_user(self, all_events, user):
         """Filters events that `user` is authorized to access"""
         if self._is_city_employee(user):
             if not settings.NEXT_PUBLIC_MOCK_FLAG:
-                if user.is_superuser:
-                    return all_events
```

Comment:
Was it your intention to remove the superuser having access to all events here?

Reply:
Yes. It still needs to be specified further whether superuser should have that right.